pkg/internal/client/txn.go

```diff
@@ -147,13 +151,23 @@ func (txn *Txn) InternalSetPriority(priority int32) {
 // transaction.
 func (txn *Txn) SetSystemConfigTrigger() {
 	txn.systemConfigTrigger = true
+	// The system-config trigger must be run on the system-config range which
+	// means any transaction with the trigger set needs to be anchored to the
+	// system-config range.
+	txn.SetTxnAnchorKey(keys.SystemConfigSpan.Key)
 }
 
 // SystemConfigTrigger returns the systemConfigTrigger flag.
 func (txn *Txn) SystemConfigTrigger() bool {
 	return txn.systemConfigTrigger
 }
 
+// SetTxnAnchorKey sets the key at which to anchor the transaction record. The
+// transaction anchor key defaults to the first key written in a transaction.
+func (txn *Txn) SetTxnAnchorKey(key roachpb.Key) {
+	txn.txnAnchorKey = key
```

Comment:
We must panic here if the transaction has already had `BeginTransaction` invoked.